python/pyarrow/tests/test_udf.py

```diff
@@ -0,0 +1,414 @@
+        assert "callback must be a callable" == execinfo.value
+
+    # validate output type
+    with pytest.raises(ValueError) as execinfo:
+        register_scalar_function(func_name, 1, doc, in_types,
+                                 None, unary_scalar_function)
+        assert "Output value type must be defined" == execinfo.value
+
+    # validate input type
+    with pytest.raises(ValueError) as execinfo:
+        register_scalar_function(func_name, 1, doc, None,
+                                 out_type, unary_scalar_function)
+        assert "input types must be of type InputType" == execinfo.value
+
+
+def test_varargs_function_validation():
+    def n_add(*values):
+        base_val = values[:2]
+        res = pc.call_function("add", base_val)
+        for other_val in values[2:]:
+            res = pc.call_function("add", [res, other_val])
+        return res
+
+    func_name = "n_add"
+    arity = 2
+    in_types = [InputType.array(pa.int64()), InputType.array(pa.int64())]
+    out_type = pa.int64()
+    doc = get_function_doc("n add function", "add N number of arrays",
+                           ["value1", "value2"])
+    register_scalar_function(func_name, arity, doc,
+                             in_types, out_type, n_add)
+
+    func = pc.get_function(func_name)
+
+    assert func.name == func_name
+
+    with pytest.raises(pa.lib.ArrowInvalid) as execinfo:
+        pc.call_function(func_name, [pa.array([1, 10]),
+                                     ])
+        error_msg = "VarArgs function 'n_add' needs at least 2 arguments"
+        +" but attempted to look up kernel(s) with only 1"
+        assert error_msg == execinfo.value
+
+
+def test_function_doc_validation():
+
+    def unary_scalar_function(scalar):
+        return pc.call_function("add", [scalar, 1])
+
+    # validate arity
+    arity = 1
+    in_types = [InputType.scalar(pa.int64())]
+    out_type = pa.int64()
+
+    # doc with no summary
+    func_doc = {}
+    func_doc["description"] = "desc"
+    func_doc["arg_names"] = ["scalar1"]
+    
+    with pytest.raises(ValueError) as execinfo:
+        register_scalar_function("no_summary", arity, func_doc, in_types,
+                                    out_type, unary_scalar_function)
+        expected_expr = "must contain summary, arg_names and a description"
+        assert expected_expr in execinfo.value
+        
+    # doc with no decription
+    func_doc = {}
+    func_doc["summary"] = "test summary"
+    func_doc["arg_names"] = ["scalar1"]
+    
+    with pytest.raises(ValueError) as execinfo:
+        register_scalar_function("no_desc", arity, func_doc, in_types,
+                                    out_type, unary_scalar_function)
+        expected_expr = "must contain summary, arg_names and a description"
+        assert expected_expr in execinfo.value
+        
+    # doc with no arg_names
+    func_doc = {}
+    func_doc["summary"] = "test summary"
+    func_doc["description"] = "some test func"
```

Comment:
General note, but just use `{"summary": ...}`. This feels like C++ code that is being translated into Python instead of Python code.